ci/ray_ci/tester.py

```diff
@@ -74,40 +74,36 @@ def main(
     os.chdir(bazel_workspace_dir)
     docker_login(_DOCKER_ECR_REPO.split("/")[0])
 
-    if not build_name:
-        build_name = f"{team}build"
-    container = TesterContainer(build_name)
+    container = _get_container(
+        team, workers, worker_id, parallelism_per_worker, build_name
+    )
     if run_flaky_tests:
         test_targets = _get_flaky_test_targets(team)
     else:
         test_targets = _get_test_targets(
             container,
             targets,
             team,
-            workers,
-            worker_id,
             except_tags,
         )
-    success = container.run_tests(test_targets, test_env, parallelism_per_worker)
+    success = container.run_tests(test_targets, test_env)
     sys.exit(0 if success else 1)
 
 
-def _get_test_targets(
-    container: TesterContainer,
-    targets: str,
+def _get_container(
     team: str,
     workers: int,
     worker_id: int,
-    except_tags: Optional[str] = "",
-    yaml_dir: Optional[str] = None,
-) -> List[str]:
-    """
-    Get test targets to run for a particular shard
-    """
-    return shard_tests(
-        _get_all_test_targets(container, targets, team, except_tags, yaml_dir=yaml_dir),
-        workers,
-        worker_id,
+    parallelism_per_worker: int,
+    build_name: Optional[str] = None,
+) -> TesterContainer:
+    return TesterContainer(
+        build_name or f"{team}build",
```

Comment:
I feel that this is a bit too magical.

I can accept if it is using "forge" as the default. but using `{team}build` feels a bit surprising. I would rather keep it explicit.

Reply:
maybe i can move the default build name into this config https://github.com/ray-project/ray/blob/master/ci/ray_ci/core.tests.yml? The concept of 'team' in our ci is pretty neat to use as an index to derive many other things (flaky tests, test tag filtering), to reduce the number of flags that this binary needs.

Reply:
I can do this in another PR though, this part in this PR is only a refactoring not a new logic